server/budgettracker.business/Api/BudgetApi.cs

```diff
@@ -110,6 +111,27 @@ public async Task<ApiResponse> DeleteBudgets(ApiRequest request)
             return response;
         }
 
+        public async Task<ApiResponse> GetBudget(ApiRequest request)
+        {
+            await Authenticate(request);
+
+            GetBudgetArgumentApiContract getBudget = request.Arguments<GetBudgetArgumentApiContract>();
+
+            ApiResponse response = new ApiResponse();
+
+            try
+            {
+                Budget retrievedBudget = await _budgetRepository.GetBudget(getBudget.BudgetValues.Id);                
```

Comment:
I think we should add a check in the API method to make sure the budget is actually the authenticated user's budget. That way they can't just ask for someone elses budget.

Reply:
Acting like you ain't honeymooning. haha

Also can do ill add a userId check to confirm they belong to the authenticated user.